packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -221,10 +221,9 @@ class CupertinoSheetTransition extends StatefulWidget {
       reverseCurve: reverseCurve,
       parent: secondaryAnimation,
     );
-    final double deviceCornerRadius = MediaQuery.maybeViewPaddingOf(context)?.top ?? 0;
```

Comment:
Lets keep this since it lets us smoothly animate from rounded device corners. This is an approximation though, and from testing I think this is not quite right. Lowering it seems to smooth out the end of the transition. So I think this is better:
``` dart
final double deviceCornerRadius = (MediaQuery.maybeViewPaddingOf(context)?.top ?? 0) * .9;
```

Paired with that, we do not want the same transition if the device corner radius is too small. In these cases we want the transition to start from zero.

So I added:
``` dart
final bool roundedDeviceCorners = deviceCornerRadius > 20;
```

Combined with the `0.9` smoothing, this seems like a good cutoff. Roughly, it equals about double the targeted radius of `12`.